src/plugins/elasticsearch/lib/map_uri.js

```diff
@@ -1,6 +1,18 @@
 import querystring from 'querystring';
 import { resolve } from 'url';
+import _ from 'lodash';
+
 module.exports = function mapUri(server, prefix) {
+
+  const filterHeaders = function (originalHeaders) {
+    const headersToKeep = server.config().get('elasticsearch.requestHeaders');
+    const headersToKeepNormalized = headersToKeep.map(function (header) {
+      return header.trim().toLowerCase();
+    });
+
+    return _.pick(originalHeaders, headersToKeepNormalized);
```

Comment:
I'm not sure if `originalHeaders` will always come back lower-case or not. I think it's probably best to normalize them as well when filtering the list.